jerry-core/ecma/operations/ecma-reference.c

```diff
@@ -132,38 +183,66 @@ ecma_op_resolve_reference_value (ecma_object_t *lex_env_p, /**< starting lexical
       ecma_object_t *binding_obj_p = ecma_get_lex_env_binding_object (lex_env_p);
 
 #if ENABLED (JERRY_LCACHE)
-      ecma_property_t *property_p = ecma_lcache_lookup (binding_obj_p, name_p);
-
-      if (property_p != NULL)
+#if ENABLED (JERRY_ES2015)
+      /* Global environment record does not have withEnvironment flag,
+         therefore the direct lcache lookup can be performed.  */
+      if (lex_env_p == ecma_get_global_environment ())
```

Comment:
If I understand correctly only the `if` check is for ES2015 any other thing is the same in all profiles?
If so what if we do something like this?:
```c
#if ENABLED(..)
bool lcache_lookup_allowed = (lex_env_p == ecma_get_global_environment ());
#else
bool lcache_lookup_allowed = true;
#endif

if (lcache_lookup_allowed)
{
```

This way we could avoid guarding the closing brackets.

What does the other reviewers think about this? @dbatyai @rerobika ?

Reply:
I second this approach.